hibernate-core/src/main/java/org/hibernate/type/ComponentType.java

```diff
@@ -64,10 +64,9 @@ public class ComponentType extends AbstractType implements CompositeTypeImplemen
 
 	private final boolean isKey;
 	private boolean hasNotNullProperty;
-	private final boolean createEmptyCompositesEnabled;
 	private final CompositeUserType<Object> compositeUserType;
 
-	private EmbeddableValuedModelPart mappingModelPart;
+	private transient EmbeddableValuedModelPart mappingModelPart;
```

Comment:
not sure if we want to Serialize also this field, the `ComponentType` is used by the cache key for checking the equality so the involved fields are just `propertyTypes` and `compositeUserType`.